packages/models/src/azure-openai/chat.ts

```diff
@@ -0,0 +1,282 @@
+import { POST, HttpError } from '@axflow/models/shared';
+import { streamTransformer } from '../openai/shared';
+
+export function headers(apiKey: string, customHeaders?: Record<string, string>) {
+  const headers: Record<string, string> = {
+    accept: 'application/json',
+    'content-type': 'application/json',
+    ...customHeaders,
+  };
+
+  if (typeof apiKey === 'string') {
+    headers['api-key'] = apiKey;
+  }
+
+  return headers;
+}
+
+// Our currently supported API version. Swagger spec:
```

Comment:
The stable version `2023-05-15` does not support function calling, which I consider important enough to bump to their preview branch.